### ISSUE TITLE
Introduce Sequence concept as a collection of Events

### ISSUE THREAD

**Issue by JJJ** [sugarcalendar/core#74]
Recurring Events ultimately consist of an ordered series of events, all relating back to a single source instance of an Event object. This series of Events currently has no way of being calculated on a per-Event basis. Rather, they are built just-in-time based on the current database query and/or Calendar view boundaries.

It should be possible, given any number of Events or any boundaries, for these Sequences of Events to be built up relative to either the initial instance or the boundary.

In short, a Sequence is a Collection of objects, though each object in the Sequence ends up with a unique identifier that is calculable relative to its source.

**Comment by JJJ** [comment]
This is largely done in the Recur class.
